lib/waypointSenderFunctionsLib.cpp

```diff
@@ -69,23 +69,17 @@ geometry_msgs::PoseStamped set_heading(float heading, float local_desired_headin
   return waypoint;
 }
 
-geometry_msgs::PoseStamped set_destination(float x, float y, float z, float psi, float correction_heading, float local_desired_heading, geometry_msgs::Pose correction_vector, geometry_msgs::Point local_offset_pose, geometry_msgs::PoseStamped waypoint)
+geometry_msgs::PoseStamped set_destination(float x, float y, float psi, geometry_msgs::PoseStamped waypoint)
 {
-	waypoint = set_heading(psi, local_desired_heading, correction_heading, waypoint);
 
 	//transform map to local
 	float deg2rad = (M_PI/180);
-	float Xlocal = x*cos((correction_heading - 90)*deg2rad) - y*sin((correction_heading - 90)*deg2rad);
-	float Ylocal = x*sin((correction_heading - 90)*deg2rad) + y*cos((correction_heading - 90)*deg2rad);
-	float Zlocal = z;
-
-	x = Xlocal + correction_vector.position.x;
-	y = Ylocal + correction_vector.position.y;
-	z = Zlocal + correction_vector.position.z;
+	float Xlocal = x*cos((90)*deg2rad) - y*sin((90)*deg2rad);
+	float Ylocal = x*sin((90)*deg2rad) + y*cos((90)*deg2rad);
 
-	waypoint.pose.position.x = -x;
-	waypoint.pose.position.y = -y;
-	waypoint.pose.position.z = z;
+	waypoint.pose.position.x = Xlocal;
```

Comment:
Why are the negatives removed here? Were they even supposed to be there originally I just forget.

Reply:
Sorry meant to mention this to you, I figured out why we needed to add these negatives. There was a mistake in what the original writer of this code had. If you look above I changed the -90 in the formula to 90 and that fixed the issue. Basically the waypoints need to be rotated 90 degrees to match Gazebo but they were being rotated the wrong way.